wagtail/core/blocks/field_block.py

```diff
@@ -545,6 +558,9 @@ def get_prep_value(self, value):
         # the JSONish representation
         return value.source
 
+    def get_form_state(self, value):
+        return value.source
```

Comment:
This method should be returning the value that's directly usable/renderable by the client-side component, which for Draftail is the ContentState representation. As it stands, this block saves to the database correctly but doesn't re-render the value when the form is reshown.

Looking at the rich text widget code, I think we should be calling the widget's `format_value` method here - and since that's a core Django method that could come into play on any form widget, that should happen on FieldBlock.

Reply:
This should be fixed now. But there's something weird going on with images. For some reason, it saves images with the ``<img>`` tag in the HTML, and this doesn't get converted back into the contentstate in this branch, but it does on master.

No idea what's causing this, I had a poke around the internals and found there isn't an element handler for ``<img>`` and the ``ImageElementHandler`` is registered against ``<embed>``.

Are there any differences in the rich text configuration on the telepath branches?

Reply:
Now fixed in 7c1bba6be7bd76383815671a9fc74bea941f062e - `FieldBlock.get_form_state` now passes the block value through both `block.value_for_form` (which retrieves the .source of the RichText value) and `widget.format_value` (which performs the dbHTML -> contentState conversion). Missing the first step meant that the contentState conversion was being applied to the front-end-rendered version of the field instead.